app/models/framework/definition/ast/semantic_checker.rb

```diff
@@ -77,12 +77,24 @@ def raise_when_lookup_reference_invalid(field)
         end
 
         def raise_when_management_field_invalid(info)
-          management_charge_modifier = [info.dig(:column_based, :column_names), info.dig(:flat_rate, :column)].compact.first(&:present?)
-          return if management_charge_modifier.blank?
+          of_columns_for(info).each { |referenced_field_name| validate_management_field(referenced_field_name) }
+        end
 
-          Array(management_charge_modifier).each do |referenced_field_name|
-            validate_management_field(referenced_field_name)
-          end
+        def of_columns_for(info)
+          Array(
+            if info[:column_based]
+              varies_by = Array(info[:column_based][:column_names])
+
+              percentage_columns = info[:column_based][:value_to_percentage]
```

Comment:
Is it worth either renaming the first `percentage_column` variable to something else *or* chaining this onto the following lines..?

Reply:
I had it chained originally. Rubocop complained about line length and indentation so eventually I found that it *wouldn't* complain about mutable state...

Reply:
I have re-chained it